CI slow tests fail: RuntimeError: min(): Expected reduction dim to be specified for input.numel() == 0

CI slow tests fail: https://github.com/huggingface/trl/actions/runs/26245674597/job/77243248504
> RuntimeError: min(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.

```python
FAILED tests/test_grpo_trainer.py::TestGRPOTrainerSlow::test_train_with_transformers_paged[trl-internal-testing/tiny-LlamaForCausalLM-3.2] - RuntimeError: min(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.
FAILED tests/test_grpo_trainer.py::TestGRPOTrainerSlow::test_train_with_transformers_paged[trl-internal-testing/tiny-MistralForCausalLM-0.2] - RuntimeError: min(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.
```

Stacktrace:
```python
>       trainer.train()

tests/test_grpo_trainer.py:2981: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.11/site-packages/transformers/trainer.py:1427: in train
    return inner_training_loop(
.venv/lib/python3.11/site-packages/transformers/trainer.py:1509: in _inner_training_loop
    self._run_epoch(
.venv/lib/python3.11/site-packages/transformers/trainer.py:1737: in _run_epoch
    tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
trl/trainer/grpo_trainer.py:1129: in training_step
    output = super().training_step(model, inputs, num_items_in_batch)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.11/site-packages/transformers/trainer.py:1903: in training_step
    inputs = self._prepare_inputs(inputs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
trl/extras/profiling.py:202: in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
trl/trainer/grpo_trainer.py:1158: in _prepare_inputs
    generation_batch = self._generate_and_score_completions(generation_batch)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
trl/trainer/grpo_trainer.py:1875: in _generate_and_score_completions
    ) = self._generate(prompts)
        ^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <trl.trainer.grpo_trainer.GRPOTrainer object at 0x7fc53911b290>
prompts = ["Although that way may not be obvious at first unless you're", "Although that way may not be obvious at first unless you're", "Although that way may not be obvious at first unless you're"]

    def _generate(self, prompts: list):
        device = self.accelerator.device

...

>       self._metrics[mode]["completions/min_length"].append(agg_completion_lengths.float().min().item())
                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       RuntimeError: min(): Expected reduction dim to be specified for input.numel() == 0. Specify the reduction dim with the 'dim' argument.

trl/trainer/grpo_trainer.py:1782: RuntimeError
```
```python
----------------------------- Captured stdout call -----------------------------
Returning results of generate_batch despite unexpected termination.
----------------------------- Captured stderr call -----------------------------

...

2026-05-21 18:46:41,304 - ContinuousBatchingLogger - ERROR - Error in generation loop: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.
Traceback (most recent call last):
  File "/__w/trl/trl/.venv/lib/python3.11/site-packages/transformers/generation/continuous_batching/continuous_api.py", line 847, in _run_generation_loop
    batch_processor.reset()
  File "/__w/trl/trl/.venv/lib/python3.11/site-packages/transformers/generation/continuous_batching/continuous_api.py", line 252, in reset
    self.inputs_and_outputs.reset()
  File "/__w/trl/trl/.venv/lib/python3.11/site-packages/transformers/generation/continuous_batching/input_outputs.py", line 306, in reset
    self._reset_static_tensors(full_reset=True)
  File "/__w/trl/trl/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 124, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/__w/trl/trl/.venv/lib/python3.11/site-packages/transformers/generation/continuous_batching/input_outputs.py", line 273, in _reset_static_tensors
    self._bulk_input_tensor[: self.static_inputs, : q_len + 1].zero_()
RuntimeError: Inplace update to inference tensor outside InferenceMode is not allowed.You can make a clone to get a normal tensor before doing inplace update.See https://github.com/pytorch/rfcs/pull/17 for more details.

  0%|          | 0/51 [00:00<?, ?it/s]
                                      
2026-05-21 18:46:42,305 - ContinuousBatchingLogger - ERROR - Generation thread terminated unexpectedly.

  0%|          | 0/51 [00:01<?, ?it/s]2026-05-21 18:46:42,683 - ContinuousBatchingLogger - ERROR - Requests ['req_0', 'req_1', 'req_2'] not found in results.
```

--- a/trl/trainer/grpo_trainer.py
+++ b/trl/trainer/grpo_trainer.py
@@ -1374,12 +1374,11 @@ class GRPOTrainer(_BaseTrainer):
                     unwrapped_model.to(torch.float16)
                 if self.args.cast_lm_head_to_fp32:
                     unwrapped_model.lm_head.to(torch.float32)
-                with torch.inference_mode():
-                    # Continuous batching API expects 'inputs' arg only
-                    all_outputs = unwrapped_model.generate_batch(
-                        prompt_ids, generation_config=self.generation_config, progress_bar=False
-                    )
-                    unwrapped_model.train()  # restore training mode, as generate_batch forces eval mode
+                # Continuous batching API expects 'inputs' arg only
+                all_outputs = unwrapped_model.generate_batch(
+                    prompt_ids, generation_config=self.generation_config, progress_bar=False
+                )
+                unwrapped_model.train()  # restore training mode, as generate_batch forces eval mode
             completion_ids = [output.generated_tokens for output in all_outputs.values()]
             logprobs = None  # not used in this case
 

--- a/trl/trainer/rloo_trainer.py
+++ b/trl/trainer/rloo_trainer.py
@@ -991,12 +991,11 @@ class RLOOTrainer(_BaseTrainer):
                     unwrapped_model.to(torch.bfloat16)
                 elif self.args.fp16:
                     unwrapped_model.to(torch.float16)
-                with torch.inference_mode():
-                    # Continuous batching API expects 'inputs' arg only
-                    all_outputs = unwrapped_model.generate_batch(
-                        prompt_ids, generation_config=self.generation_config, progress_bar=False
-                    )
-                    unwrapped_model.train()  # restore training mode, as generate_batch forces eval mode
+                # Continuous batching API expects 'inputs' arg only
+                all_outputs = unwrapped_model.generate_batch(
+                    prompt_ids, generation_config=self.generation_config, progress_bar=False
+                )
+                unwrapped_model.train()  # restore training mode, as generate_batch forces eval mode
             completion_ids = [output.generated_tokens for output in all_outputs.values()]
 
         else:

